lantiq: switch-setup, wifi mac fix for ARV752DPW

lrswss opened the pull request:
Few fixes for Arcadyan ARV752DPW (aka Easybox 802):

- set wifi mac from flash partition 'board_config' to match the label
on the device; currently wifi mac is always set to 00:0c:43:28:60:00
and eth0 is configured with preset mac from 'board_config'
- set eth0 mac address to be an increment of wifi mac
- add missing switch config (4 LAN-Ports + CPU)

adrianschmutzler commented:
> set eth0 mac address to be an increment of wifi mac

Why?

lrswss commented:
There's no other mac address stored in 'board_config' and I thought it's a good idea to assign individual but still vendor specific mac addresses to both interfaces as seen on other devices. Unfortunately I don't have a box with the original firmware (Vodafone) at hand to check with 'ground truth'. However I just found a few online references searching for 'easybox 802 00:23:08' (people posting full specs of their box in forums) that actually seem to underpin that in the original firmware wifi and eth0 are in deed using the same mac whereas the wan interface (grey dsl port) is assigned a mac with +1 increment. The wan port on the arv752dpw doesn't seem to work though (shows up in switch config but has no link if cable is connected), that's why I removed it with the added switch config in '01_network'.

adrianschmutzler commented:
I think it's a good idea to stick to vendor behavior for the MAC stuff. This will ensure that there are no MAC collisions if devices with consecutive addresses are used in the same network out in the field. There, I'd prefer to have the same MAC address for ethernet and WiFi if that's what the vendor does.

blocktrron commented on target/linux/lantiq/files-4.19/arch/mips/boot/dts/lantiq/danube_arcadyan_arv752dpw.dts:
```diff
@@ -157,6 +159,7 @@
 &gsw {
 	phy-mode = "rmii";
 	mtd-mac-address = <&boardconfig 0x16>;
+	mtd-mac-address-increment = <1>;
```
I'm fine with adding the switch configuration as well as using a unique MAC address for the WiFi interface.

Regarding the LAN mac address assignment I'm with @adrianschmutzler in this matter. The LAN and Wireless MAC are identical in the vendor firmware. At least this seems to be the case with the similar EasyBox 803 (see picture 5 of this [post ](https://www.computerbase.de/forum/threads/repeater-einrichten.1818328/#post-21609934)).

We should try to resemble the stock behavior where this is feasible. This is obviously the case here.

lrswss replied:
Ok, the vendor is always right. ;-) Removed the proposed increment on ethernet mac as requested.

blocktrron commented:
@lrswss The WiFi MAC fix seems to be gone now?

lrswss commented:
> @lrswss The WiFi MAC fix seems to be gone now?

The PR now consits of two consecutive commits: the most recent, second commit 83202bb just removes - as requested - the mac increment intruduced with the first commit 61c5f2d. Or is mandatory to combine all changes as a single commit? I'm not a friend of resetting already published commits.

adrianschmutzler commented:
> The PR now consits of two consecutive commits: the most recent, second commit [83202bb](https://github.com/openwrt/openwrt/commit/83202bb63c4607b418bce9e6ce02a87a9a99ddeb) just removes - as requested - the mac increment intruduced with the first commit [61c5f2d](https://github.com/openwrt/openwrt/commit/61c5f2d5928e350c9ffb24fa270fb3f2e99a4716). Or is mandatory to combine all changes as a single commit? I'm not a friend of resetting already published commits.

Well, just see what you do here as a draft. It's only "published" when it's added to the openwrt master.

So, I prefer a single, clean commit added to master if I still can change it (which is the case here). So, please squash it, as it's not a second change, but a correction of your first commit.

lrswss commented:
Ok, here you go: 8d07eae

adrianschmutzler commented:
@lrswss Content looks fine now.

Looking at it, actually, I'd prefer to split wifi and switch changes. Can you split this into two commits (should be quite trivial), I'd merge it then.

lrswss committed:
lantiq: wifi mac fix for ARV752DPW

- set wifi mac from flash partition 'board_config' to
  match the label on the device (Easybox 802); currently
  the wifi mac is always set to 00:0c:43:28:60:00; eth0 and
  wifi will now share the same mac (as in original firmare)

Signed-off-by: Lars Wessels <software@bytebox.org>

lrswss committed:
lantiq: added switch-config for ARV752DPW

- add missing switch config (4 LAN-Ports + CPU)
  grey WAN port (used for ADSL) is excluded

Signed-off-by: Lars Wessels <software@bytebox.org>

lrswss commented:
> Looking at it, actually, I'd prefer to split wifi and switch changes. Can you split this into two commits (should be quite trivial), I'd merge it then.

Ok, just did a force update of my branch for the ARV752DPW splitting the last commit into two (wifi mac fix, switch config). Everything in line now?

blocktrron reviewed (approved):
Merged with minor changes (commit message) to my staging tree.

Thanks for your contribution!

adrianschmutzler commented:
Thanks.
